src/utils/http.rs

```diff
@@ -24,57 +25,51 @@ lazy_static::lazy_static! {
     ].into_iter().map(|x| x.parse().unwrap()).collect();
 }
 
-pub fn follow_redirects<F>(
+pub async fn follow_redirects<F>(
     initial_url: Url,
     max_redirects: usize,
     make_request: F,
-) -> ResponseFuture<ClientResponse, SendRequestError>
+) -> Result<ClientResponse, SendRequestError>
 where
     F: Fn(&str) -> Result<ClientRequest, actix_web::error::Error> + Send + 'static,
 {
-    let state = (initial_url, max_redirects, true);
-    Box::new(future::loop_fn(state, move |(url, redirects, trusted)| {
-        let mut request = tryf!(make_request(url.as_str()).map_err(|err| {
+    let mut url = initial_url;
+    let mut redirects = max_redirects;
+    let mut trusted = true;
+
+    let response = loop {
+        let mut request = make_request(url.as_str()).map_err(|err| {
             let message = format!("Failed creating request: {} (URI: {})", err, url);
             sentry::capture_message(&message, sentry::Level::Error);
             SendRequestError::Connector(ClientConnectorError::InvalidUrl)
-        }));
-
+        })?;
         if !trusted {
             let headers = request.headers_mut();
             headers.remove(header::AUTHORIZATION);
             headers.remove(header::COOKIE);
         }
-
-        let resp_fut = request.send().and_then(move |response| {
-            if response.status().is_redirection() && redirects > 0 {
-                let location = response
-                    .headers()
-                    .get(header::LOCATION)
-                    .and_then(|l| l.to_str().ok());
-
-                if let Some(location) = location {
-                    let redirect_url = url.join(location).map_err(|e| {
-                        let message = format!("bad request uri: {}", e);
-                        SendRequestError::Io(io::Error::new(io::ErrorKind::InvalidData, message))
-                    })?;
-
-                    log::trace!("Following redirect: {:?}", &redirect_url);
-                    let is_same_origin = redirect_url.origin() == url.origin();
-
-                    return Ok(future::Loop::Continue((
-                        redirect_url,
-                        redirects - 1,
-                        trusted && is_same_origin,
-                    )));
-                }
+        let response = request.send().compat().await?;
```

Comment:
Can you restore all the blank lines, please? Mostly, this helps with consistency in this code base. Personally, blank lines make it much easier for me to read the code, as it clearly separates blocks and statements that server a unique purpose.